From 152f82d69308b96b0e838918387fa70c26a52454 Mon Sep 17 00:00:00 2001
From: Chris Morgan <macromorgan@hotmail.com>
Date: Wed, 2 Feb 2022 10:43:32 -0600
Subject: [PATCH] overlays: Add Geekworm mzp280 to vc4-kms-dpi-panel

Add support for the Geekworm mzp280 DPI panel to the generic
vc4-kms-dpi-panel overlay.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 arch/arm/boot/dts/overlays/README                        | 1 +
 arch/arm/boot/dts/overlays/vc4-kms-dpi-panel-overlay.dts | 7 +++++++
 2 files changed, 8 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -3682,6 +3682,7 @@ Info:   Enable a preconfigured KMS DPI p
 Load:   dtoverlay=vc4-kms-dpi-panel,<param>=<val>
 Params: at056tn53v1             Enable an Innolux 5.6in VGA TFT
         kippah-7inch            Enable an Adafruit Kippah with 7inch panel.
+        mzp280                  Enable a Geekworm MZP280 panel.
         backlight-gpio          Defines a GPIO to be used for backlight control
                                 (default of none).
         backlight-pwm           Defines a PWM channel to be used for backlight
--- a/arch/arm/boot/dts/overlays/vc4-kms-dpi-panel-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dpi-panel-overlay.dts
@@ -23,6 +23,12 @@
 			compatible = "ontat,yx700wv03", "simple-panel";
 		};
 	};
+	fragment@2 {
+		target = <&panel>;
+		__dormant__  {
+			compatible = "geekworm,mzp280", "simple-panel";
+		};
+	};
 
 	fragment@90 {
 		target = <&dpi>;
@@ -58,5 +64,6 @@
 	__overrides__ {
 		at056tn53v1 = <0>, "+0+90";
 		kippah-7inch = <0>, "+1+91";
+		mzp280 = <0>, "+2+93";
 	};
 };
